cpp/linear_regression/avocado_price_prediction/avocado_price_prediction.cpp

```diff
@@ -0,0 +1,140 @@
+/**
+* Predicting Avocado's Average Price using Linear Regression
+* Our target is to predict the future price of avocados depending on various
+* features (Type, Region, Total Bags, ...).
+*
+* Dataset
+* 
+* Avocado Prices dataset has the following features:
+* PLU - Product Lookup Code in Hass avocado board.
+* Date - The date of the observation.
+* AveragePrice - Observed average price of single avocado.
+* Total Volume - Total number of avocado's sold.
+* 4046 - Total number of avocado's with PLU 4046 sold.
+* 4225 - Total number of avocado's with PLU 4225 sold.
+* 4770 - Total number of avocado's with PLU 4770 sold.
+* Total Bags = Small Bags + Large Bags + XLarge Bags.
+* Type - Conventional or organic.
+* Year - Year of observation.
+* Region - City or region of observation.
+*
+* Approach
+* 
+* In this example, first we will do EDA on the dataset to find correlation
+* between various features.
+* Then we'll be using onehot encoding to encode categorical features.
+* Finally we will use LinearRegression API from mlpack to learn the correlation
+* between various features and the target i.e AveragePrice.
+* After training the model, we will use it to do some predictions, followed by
+* various evaluation metrics to quantify how well our model behaves.
+*/
+#include <mlpack.hpp>
+using namespace mlpack;
+using namespace mlpack::data;
```

Comment:
```suggestion
#include <mlpack.hpp>

using namespace mlpack;
```

I don't think that `mlpack::data` is actually being used (I see calls to `data::DatasetInfo` later, for instance).